README.md

```diff
@@ -27,6 +28,42 @@ This repository contains the code of the backend used by the [web](https://githu
 
 ---
 
+## What is this?
+
+This is the backend that supports the [italia-app](https://github.com/teamdigitale/italia-app)
+mobile application.
+
+This project is part of the Italian Digital Citizenship initiative, see the
+[main repository](https://github.com/teamdigitale/digital-citizenship) for further information.
+
+## Authentication process
+
+The `italia-app` application will authenticate to the backend in two steps:
+
+  1. an initial user initiated SPID authentication process (SAML2 based)
+     that identifies the user and, on success, triggers the creation of a new
+     authentication session (associated to a session token)
+  2. subsequent requests to the backend will be authenticated via a bearer session token
+  
+See the sequence diagram [here](https://www.planttext.com/?text=PO_H2e9044NVzolc1w9teJ9KQa5qqUrPH8TOqDrbTepy-xOY47eSEFVSEsdWddYP9o7EtiFk28hSSqtWKcZOJ-vkx0aj6rvbVZ2Zxxv7PqOXFkGx8dELm5Lh1OIZ8HmOs44aQLNEM1OY4XktgjiexqD0OkDivcnOPgGMwM46t8X_Mz0YzOomn0G4326cDbXS_jVl8IjbKUckANBP59cMTTNTcbA85EqO7tu1)
+
+### User authentication
+
+When a client wants to login with the backend it will call the `/login` endpoint with the IDP entityID as parameter in
+the query string. The backend will then build an authorization URL and performs a redirect to the chosen IDP. The
+authentication process will continue to the IDP website. If the authentication process ends with success the IDP will
+redirect the client to an HTML page with a form that will auto-post itself to the `/assertionConsumerService` endpoint
+with a SAMLResponse as an hidden field. The backend will then parse and validate the SAMLResponse to extract all the
+user attributes, then it generates an unique token and saves an User object to the `SessionStorage` service using the
```

Comment:
what's inside this User object ? (is a stateful session necessary ?)

Reply:
the User object contains the attribute returned by the SPID authentication (fiscal code, first name, last name, email)

Reply:
yes it's basically the user data we get from SPID - the other option we have is to keep the application server stateless and store everything in a fat session token - the downside would be unnecessary amount of data being sent by the app over mobile